nextstrain/cli/command/login.py

```diff
@@ -0,0 +1,153 @@
+"""
+Logs in a Nextstrain.org groups user, storing their credentials locally.
+"""
+import os
+import ast
+import sys
+import jose
+import boto3
+import getpass
+import configparser
+from pathlib import Path
+from pycognito import Cognito
+from pycognito.aws_srp import AWSSRP
+from typing import Dict
+
+base = Path(__file__).resolve().parent.parent.parent.parent
+secrets_path = base / '.nextstrain/secrets'
+config = configparser.ConfigParser()
+
+def register_parser(subparser):
+    parser = subparser.add_parser("login", help = "Login as a Nextstrain.org groups user")
+
+    register_arguments(parser)
+
+    return parser
+
+
+def register_arguments(parser):
+    parser.add_argument(
+        "--check",
+        help    = "Check credentials by logging into nextstrain.org",
+        action  = 'store_true')
+
+    parser.add_argument(
+        "--delete",
+        help    = "Deletes the secrets file containing authentication credentials",
+        action  = 'store_true')
+
+    return parser
+
+
+def run(opts):
+    if opts.delete:
+        try:
+            os.remove(secrets_path)
+            print(f"Deleted the credentials file at {secrets_path}.")
+
+        except FileNotFoundError:
+            print("No credentials found to delete!")
+
+        sys.exit(0)
+
+    if opts.check:
+        print("Checking credentials by logging into nextstrain.org...")
+        try:
+            config.read(secrets_path)
+            username = config.get('login', 'username')
+            authn_result = ast.literal_eval(config.get('login', 'authenticationresult'))
```

Comment:
🤨 This makes me raise an eyebrow as unnecessary complexity/use of the AST parser. Instead of storing a stringified dict, why not store the tokens under individual keys (e.g. `config.get("id_token")`, `config.get("refresh_token")`, …)?

Reply:
Actually, what's storing `authenticationresult`? I don't see if referenced anywhere else.

Reply:
`authenticationresult` was one of they keys in the dict returned by the AWSSRP utility class that I in turn saved to the config. It will probably not be relevant after switching to using only the Cognito utility class.